elyra/pipeline/component_registry.py

```diff
@@ -176,52 +173,67 @@ def _read_component_registry(self) -> Tuple[List, List]:
-                        "label": category_metadata["label"],
-                        "image": category_metadata["image"],
-                        "description": category_metadata["description"]
-                    }
-                    category_entries.append(SimpleNamespace(**category_entry))
+                    category_entry = ComponentCategory(id=category_id,
+                                                       label=category_metadata.get('label'),
+                                                       image_location=category_metadata.get('image'),
+                                                       description=category_metadata.get('description'))
+                    registry['categories'].append(category_entry)
 
-        return component_entries, category_entries
+        return registry
 
-    def _get_relative_location(self, component_type: str, component_path: str):
+    def _get_relative_location(self, component_type: str, component_path: str) -> str:
         """
         Gets the absolute path for a component from a file-based registry
         """
         if component_type == "filename":
             component_path = f"{self._parser._type}/{component_path}"
         return component_path
 
-    def _get_component_registry_entry(self, component_id):
-        """
-        Get the body of the component registry entry with the given id
-        """
-        # Read registry to find list of components
-        component_entries, _ = self._read_component_registry()
-
-        # Find entry with the appropriate id, if exists
-        component_entry = next((entry for entry in component_entries if entry.id == component_id), None)
-        if not component_entry:
-            self.log.error(f"Component with ID '{component_id}' could not be found in the " +
-                           f"{self._component_registry_location} component_id catalog.")
-            raise ValueError(f"Component with ID '{component_id}' could not be found in the " +
-                             f"{self._component_registry_location} component_id catalog.")
-
-        return component_entry
+    def _read_registry_for_component(self, queried_component_id: str, catalog_entry_id: str) -> Component:
+        with open(self._component_registry_location, 'r') as catalog_file:
+            catalog_json = json.load(catalog_file)
+            component_entry = dict(catalog_json['components'].items()).get(catalog_entry_id)
+            if not component_entry:
+                self.log.error(f"Component with ID '{queried_component_id}' could not be found in the " +
+                               f"{self._component_registry_location} component_id catalog.")
+                raise ValueError(f"Component with ID '{queried_component_id}' could not be found in the " +
+                                 f"{self._component_registry_location} component_id catalog.")
+
+            component_type = next(iter(component_entry.get('location')))
```

Comment:
Is this statement extracting the first (and only) key name within the location stanza? I.e.,
```python
component_type = component_entry.get('location').keys()[0]
```

 I find the `next(iter(` stuff difficult, but that's due to my unfamiliarity with Python.  I think a comment would be helpful here.

Reply:
I just checked that the two statements are only logically equivalent (`keys()` returns a `dict_keys` object which is not scriptable), so please add a comment that you're going after the key name for the entry in `location`.

Reply:
I wonder if this variable should be named `location_type` since it's value will be either `"url"` or `"filename"`?